cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/DefaultWorkflowForkConfigurer.java

```diff
@@ -34,14 +39,16 @@
  * @param <T>
  */
 public class DefaultWorkflowForkConfigurer<T extends WorkflowForkJoiner & WorkflowConditionAdder>
-  implements WorkflowForkConfigurer<T>, WorkflowForkJoiner, WorkflowConditionAdder {
+  extends DefaultPluginConfigurer implements WorkflowForkConfigurer<T>, WorkflowForkJoiner, WorkflowConditionAdder {
```

Comment:
Seems like the extension is for the sake of having `artifactId`, `artifactRepo`... from it. You can just memorize them as fields of this class in the constructor, isn't it? Or the `WorkflowForkConfigurer` actually has the plugin configurer functionalities?